crates/api_common/src/request.rs

```diff
@@ -200,50 +205,59 @@ pub async fn fetch_site_data(
       // Warning, this may ignore SSL errors
       let metadata_option = fetch_site_metadata(client, url).await.ok();
       if !include_image {
-        return (metadata_option, None);
+        (metadata_option, None)
+      } else {
+        let thumbnail_url =
+          match fetch_pictrs_hash(client, &metadata_option, settings, url).await {
+            Ok(pictrs_hash) => Url::parse(&format!(
+              "{}/pictrs/image/{}",
+              settings.get_protocol_and_hostname(),
+              pictrs_hash
+            ))
+            .map(Into::into)
+            .map_err(LemmyError::from),
+            Err(err) => match err.error_type {
+              LemmyErrorType::PictrsCachingDisabled => Ok(<Url>::clone(url).into()),
```

Comment:
This is your bug here. You've made the thumbnail_url = the site url above, which often isn't an image. 

You can pry remove this whole section, since `fetch_pictrs` already handles it.

Reply:
NEVERMIND I cannot read, you're 100% correct 🤦